internal/controller/promotions/promotions.go

```diff
@@ -399,14 +401,6 @@ func (r *reconciler) promote(
 	}
 	logger.Debug("found associated Stage")
 
-	targetFreight, err := kargoapi.GetFreight(
-		ctx,
-		r.kargoClient,
-		types.NamespacedName{
-			Namespace: promo.Namespace,
-			Name:      promo.Spec.Freight,
-		},
-	)
```

Comment:
I think the err check and nil check below this need to be removed as well.

Reply:
Actually... I just noticed that this nil check is missing from the Reconcile method where you obtain the Freight that you pass to this function. It should be added.

Reply:
We spoke offline and I understand now that Freight being not found would have no bearing on whether you emit the event or not, so leaving the nil check here makes sense to me.